engines/stark/ui/world/topmenu.cpp

```diff
@@ -69,7 +69,7 @@ void TopMenu::onMouseMove(const Common::Point &pos) {
 		Button *hoveredButton = getButtonAtPosition(pos);
 		if (hoveredButton) {
 			_cursor->setCursorType(Cursor::kActive);
-			_cursor->setMouseHint(hoveredButton->getText());
+			_cursor->setMouseHint(hoveredButton->getText(), hoveredButton->getHintPosition());
```

Comment:
I suggest not reusing the same code as for world object mouse hints for the top menu button hints. They don't work the same way. This will make the code in the Cursor class cleaner and make it easier to implement the mouse hint delay.
The code for the top menu hints can be in the TopMenu class or in the Button class.